modules/gatsby/src/gatsby-node/create-pages.js

```diff
@@ -248,3 +220,62 @@ function createDocPages({ graphql, actions }) {
     });
   });
 }
+
+// This is a main gatsby entry point
+// Here we get to programmatically create pages after all nodes are created
+// by gatsby.
+// We use graphgl to query for nodes and iterate
+module.exports = function createPages({ graphql, actions }, pluginOptions) {
+  log.log({ color: COLOR.CYAN }, 'generating pages')();
+  const { createPage } = actions;
+  // TODO/ib - plugin options no longer provided when we are not a plugin
+  // We seem to be getting site metadata instead?
+  const {
+    docPages = true,
+    examplePages = true // TODO - autodetect based on DEMOS config
+  } = pluginOptions;
+
+  createIndexPage({ graphql, actions });
+  const searchPromise = graphql(`
```

Comment:
* Put the search query and page creation code in helper function `createSearchPage`?
* Return a promise from each `create...Page` function and do a `return Promise.all([create...]);` here in the main function?